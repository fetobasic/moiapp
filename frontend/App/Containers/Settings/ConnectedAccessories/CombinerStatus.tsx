import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { NativeStackScreenProps } from '@react-navigation/native-stack';

import { ApplicationStyles, Colors, Fonts, Metrics } from 'App/Themes';
import { HeaderSimple as Header } from 'App/Components';
import { SettingsStackParamList } from 'App/Types/NavigationStackParamList';

type Props = NativeStackScreenProps<SettingsStackParamList, 'CombinerStatus'>;

const CombinerStatus = ({ route }: Props) => {
  const isYeti20004000 = route?.params?.isYeti20004000;

  return (
    <View style={ApplicationStyles.mainContainer}>
      <Header title="Yeti Series Combiner Status" />
      <View style={styles.container}>
        <View style={styles.row}>
          <View style={styles.sectionIcon}>
            <View style={[styles.statusIcon, styles.statusIconAvailable]} />
          </View>
          <View style={styles.textSection}>
            <Text style={styles.textTitle}>120V Available</Text>
            <Text style={styles.textSubTitle}>Green</Text>
            <Text style={styles.textDescription}>
              Only one Yeti detected. Waiting for the second Yeti to be connected.
            </Text>
          </View>
        </View>
        <View style={styles.row}>
          <View style={styles.sectionIcon}>
            <View style={[styles.statusIcon, styles.statusIconSyncing]} />
          </View>
          <View style={styles.textSection}>
            <Text style={styles.textTitle}>Syncing</Text>
            <Text style={styles.textSubTitle}>Yellow</Text>
            <Text style={styles.textDescription}>Syncing connected Yetis. They will be ready shortly.</Text>
          </View>
        </View>
        <View style={styles.row}>
          <View style={styles.sectionIcon}>
            <View style={[styles.statusIcon, styles.statusIconActive]} />
          </View>
          <View style={styles.textSection}>
            <Text style={styles.textTitle}>240V / {isYeti20004000 ? '30' : '15'}A Active</Text>
            <Text style={styles.textSubTitle}>Blue</Text>
            <Text style={styles.textDescription}>
              The Yetis are combined. You can now use 240V AC outputs up to {isYeti20004000 ? '30' : '15'} Amps.
            </Text>
          </View>
        </View>
      </View>
    </View>
  );
};

export default CombinerStatus;

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: Metrics.baseMargin,
    paddingVertical: Metrics.marginSection,
    flex: 1,
  },
  row: {
    flexDirection: 'row',
    paddingVertical: Metrics.smallMargin,
    paddingHorizontal: Metrics.marginSection,
    marginBottom: Metrics.marginSection,
  },
  sectionIcon: {
    width: 32,
    height: 32,
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: Metrics.smallMargin,
  },
  statusIcon: {
    width: 12,
    height: 12,
    borderRadius: 6,
  },
  statusIconAvailable: {
    backgroundColor: Colors.combinerStatus.available,
  },
  statusIconSyncing: {
    backgroundColor: Colors.combinerStatus.syncing,
  },
  statusIconActive: {
    backgroundColor: Colors.combinerStatus.active,
  },
  textSection: {
    marginLeft: Metrics.smallMargin,
    marginRight: Metrics.baseMargin,
  },
  textTitle: {
    ...Fonts.font.base.bodyTwo,
    color: Colors.white,
  },
  textSubTitle: {
    ...Fonts.font.base.bodyTwo,
    fontSize: 10,
    lineHeight: 24,
    color: Colors.disabled,
  },
  textDescription: {
    ...Fonts.font.base.bodyOne,
    color: Colors.disabled,
  },
});
