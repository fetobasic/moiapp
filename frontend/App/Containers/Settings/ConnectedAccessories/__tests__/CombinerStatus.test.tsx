import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import { Provider } from 'react-redux';
import createMockStore from 'redux-mock-store';
import { mockedState } from 'App/Fixtures/mocks/mockedState';
import CombinerStatus from '../CombinerStatus';

const mockStore = createMockStore([]);
const store = mockStore({
  ...mockedState,
});

describe('CombinerStatus', () => {
  test('renders screen', () => {
    const props = {
      navigation: {
        navigate: jest.fn() as any,
      },
      route: { params: { isYeti20004000: true } },
    } as any;

    const tree = render(
      <Provider store={store}>
        <NavigationContainer>
          <CombinerStatus {...props} />
        </NavigationContainer>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders screen isYeti20004000: false', () => {
    const props = {
      navigation: {
        navigate: jest.fn() as any,
      },
      route: { params: { isYeti20004000: false } },
    } as any;

    const tree = render(
      <Provider store={store}>
        <NavigationContainer>
          <CombinerStatus {...props} />
        </NavigationContainer>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
