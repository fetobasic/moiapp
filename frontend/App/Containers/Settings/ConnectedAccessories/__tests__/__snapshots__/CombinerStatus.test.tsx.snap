// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CombinerStatus renders screen 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(27, 27, 27, 1)",
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(27, 27, 27, 1)",
          "borderBottomLeftRadius": 24,
          "borderBottomRightRadius": 24,
          "elevation": 12,
          "shadowColor": "rgba(0, 0, 0, 1)",
          "shadowOffset": {
            "height": 4,
            "width": 4,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 12,
          "zIndex": 1,
        },
        {
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 24,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={20}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "opacity": 1,
            },
            {
              "left": 0,
              "padding": 10,
              "position": "absolute",
            },
          ]
        }
      >
        <SvgMock />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "marginHorizontal": 48,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "RobotoCondensed-Medium",
              "fontSize": 20,
              "fontWeight": "500",
              "lineHeight": 32,
            }
          }
        >
          Yeti Series Combiner Status
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "flex": 1,
        "paddingHorizontal": 24,
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 16,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": 32,
            "justifyContent": "center",
            "marginTop": 8,
            "width": 32,
          }
        }
      >
        <View
          style={
            [
              {
                "borderRadius": 6,
                "height": 12,
                "width": 12,
              },
              {
                "backgroundColor": "rgba(3, 207, 3, 1)",
              },
            ]
          }
        />
      </View>
      <View
        style={
          {
            "marginLeft": 8,
            "marginRight": 24,
          }
        }
      >
        <Text
          style={
            {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
        >
          120V Available
        </Text>
        <Text
          style={
            {
              "color": "rgba(133, 133, 133, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 10,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
        >
          Green
        </Text>
        <Text
          style={
            {
              "color": "rgba(133, 133, 133, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
            }
          }
        >
          Only one Yeti detected. Waiting for the second Yeti to be connected.
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 16,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": 32,
            "justifyContent": "center",
            "marginTop": 8,
            "width": 32,
          }
        }
      >
        <View
          style={
            [
              {
                "borderRadius": 6,
                "height": 12,
                "width": 12,
              },
              {
                "backgroundColor": "rgba(255, 205, 29, 1)",
              },
            ]
          }
        />
      </View>
      <View
        style={
          {
            "marginLeft": 8,
            "marginRight": 24,
          }
        }
      >
        <Text
          style={
            {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
        >
          Syncing
        </Text>
        <Text
          style={
            {
              "color": "rgba(133, 133, 133, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 10,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
        >
          Yellow
        </Text>
        <Text
          style={
            {
              "color": "rgba(133, 133, 133, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
            }
          }
        >
          Syncing connected Yetis. They will be ready shortly.
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 16,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": 32,
            "justifyContent": "center",
            "marginTop": 8,
            "width": 32,
          }
        }
      >
        <View
          style={
            [
              {
                "borderRadius": 6,
                "height": 12,
                "width": 12,
              },
              {
                "backgroundColor": "rgba(28, 154, 214, 1)",
              },
            ]
          }
        />
      </View>
      <View
        style={
          {
            "marginLeft": 8,
            "marginRight": 24,
          }
        }
      >
        <Text
          style={
            {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
        >
          240V / 
          30
          A Active
        </Text>
        <Text
          style={
            {
              "color": "rgba(133, 133, 133, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 10,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
        >
          Blue
        </Text>
        <Text
          style={
            {
              "color": "rgba(133, 133, 133, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
            }
          }
        >
          The Yetis are combined. You can now use 240V AC outputs up to 
          30
           Amps.
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`CombinerStatus renders screen isYeti20004000: false 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(27, 27, 27, 1)",
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(27, 27, 27, 1)",
          "borderBottomLeftRadius": 24,
          "borderBottomRightRadius": 24,
          "elevation": 12,
          "shadowColor": "rgba(0, 0, 0, 1)",
          "shadowOffset": {
            "height": 4,
            "width": 4,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 12,
          "zIndex": 1,
        },
        {
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 24,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={20}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "opacity": 1,
            },
            {
              "left": 0,
              "padding": 10,
              "position": "absolute",
            },
          ]
        }
      >
        <SvgMock />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "marginHorizontal": 48,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "RobotoCondensed-Medium",
              "fontSize": 20,
              "fontWeight": "500",
              "lineHeight": 32,
            }
          }
        >
          Yeti Series Combiner Status
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "flex": 1,
        "paddingHorizontal": 24,
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 16,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": 32,
            "justifyContent": "center",
            "marginTop": 8,
            "width": 32,
          }
        }
      >
        <View
          style={
            [
              {
                "borderRadius": 6,
                "height": 12,
                "width": 12,
              },
              {
                "backgroundColor": "rgba(3, 207, 3, 1)",
              },
            ]
          }
        />
      </View>
      <View
        style={
          {
            "marginLeft": 8,
            "marginRight": 24,
          }
        }
      >
        <Text
          style={
            {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
        >
          120V Available
        </Text>
        <Text
          style={
            {
              "color": "rgba(133, 133, 133, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 10,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
        >
          Green
        </Text>
        <Text
          style={
            {
              "color": "rgba(133, 133, 133, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
            }
          }
        >
          Only one Yeti detected. Waiting for the second Yeti to be connected.
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 16,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": 32,
            "justifyContent": "center",
            "marginTop": 8,
            "width": 32,
          }
        }
      >
        <View
          style={
            [
              {
                "borderRadius": 6,
                "height": 12,
                "width": 12,
              },
              {
                "backgroundColor": "rgba(255, 205, 29, 1)",
              },
            ]
          }
        />
      </View>
      <View
        style={
          {
            "marginLeft": 8,
            "marginRight": 24,
          }
        }
      >
        <Text
          style={
            {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
        >
          Syncing
        </Text>
        <Text
          style={
            {
              "color": "rgba(133, 133, 133, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 10,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
        >
          Yellow
        </Text>
        <Text
          style={
            {
              "color": "rgba(133, 133, 133, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
            }
          }
        >
          Syncing connected Yetis. They will be ready shortly.
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 16,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": 32,
            "justifyContent": "center",
            "marginTop": 8,
            "width": 32,
          }
        }
      >
        <View
          style={
            [
              {
                "borderRadius": 6,
                "height": 12,
                "width": 12,
              },
              {
                "backgroundColor": "rgba(28, 154, 214, 1)",
              },
            ]
          }
        />
      </View>
      <View
        style={
          {
            "marginLeft": 8,
            "marginRight": 24,
          }
        }
      >
        <Text
          style={
            {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
        >
          240V / 
          15
          A Active
        </Text>
        <Text
          style={
            {
              "color": "rgba(133, 133, 133, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 10,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
        >
          Blue
        </Text>
        <Text
          style={
            {
              "color": "rgba(133, 133, 133, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
            }
          }
        >
          The Yetis are combined. You can now use 240V AC outputs up to 
          15
           Amps.
        </Text>
      </View>
    </View>
  </View>
</View>
`;
